pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -923,16 +923,15 @@ func getUsableDiskSize(path string) (int64, error) {
 	if err != nil {
 		return int64(-1), err
 	}
+
 	availableSize := int64(statfs.Bavail) * int64(statfs.Bsize)
 
-	var stat syscall.Stat_t
-	err = syscall.Stat(path, &stat)
-	if err != nil {
-		return int64(-1), err
-	}
-	actualSize := int64(stat.Size)
+	totalSize := int64(statfs.Blocks) * int64(statfs.Bsize)
+	usedSize := totalSize - (int64(statfs.Bfree) * int64(statfs.Bsize))
```

Comment:
Bavail, no?

Reply:
According to this: https://man7.org/linux/man-pages/man2/statfs.2.html

Bavail returns the available size for unprivileged users, this doesn't include the overhead reserved by the storage provisioner.
Bfree returns all the total free blocks from the fs, including the ones that are already reserved by the storage provisioner.

There was some of trial and error to reach this calculation but it's the one that matches the actual used size that qemu-img returns.

Reply:
Using qemu-img as discussed offline.